omnibus/config/software/datadog-agent.rb

```diff
@@ -57,6 +59,12 @@
 
   # include embedded path (mostly for `pkg-config` binary)
   env = with_standard_compiler_flags(with_embedded_path(env))
+
+  # Use msgo toolchain when fips mode is enabled
+  if fips_mode? && !windows_target?
+    env["GOROOT"] = msgoroot
+    env["PATH"] = "#{msgoroot}/bin:#{env['PATH']}"
+  end
```

Comment:
Non blocking, but I think this would be better suited in `get_omnibus_env` (in `tasks/omnibus.py`)

Reply:
👍 I will have a follow up PR for this update so we can merge this before the code freeze